ote/src/cljs/ote/views/transit_visualization.cljs

```diff
@@ -552,36 +557,22 @@
                                                                  [(dec current-year)])
                                                                [current-year]
                                                                [(inc current-year)]))
                                            :hover-style #(let [d (time/format-date-iso-8601 %)
-                                                               hover-date (goog.date.DateTime. %)
                                                                hash (date->hash d)
-                                                               date1 (goog.date.DateTime. (:date1 compare))
-                                                               date2 (goog.date.DateTime. (:date2 compare))
                                                                hash-color (hash->color hash)]
-                                                           (when-not (or (= (time/format-date-iso-8601 date1) d)
-                                                                         (= (time/format-date-iso-8601 date2) d))
-                                                             (cond (t/after? date1 hover-date)
-                                                                   (style/date1-highlight-style hash-color
-                                                                                                style/date1-highlight-color-hover)
-                                                                   (t/after? hover-date date2)
-                                                                   (style/date2-highlight-style hash-color
-                                                                                                style/date2-highlight-color-hover)
-                                                                   :else
-                                                                   (if (= 2 (get compare :last-selected-date 2))
-                                                                     (style/date1-highlight-style hash-color
-                                                                                                  style/date1-highlight-color-hover)
-                                                                     (style/date2-highlight-style hash-color
-                                                                                                  style/date2-highlight-color-hover)))))}]
-
-       [:h3 "Valittujen päivämäärien väliset muutokset"]
-       [comparison-date-changes compare]]]]))
+                                                           (style/date-highlight-style hash-color
+                                                                                       style/date-highlight-color-hover))}]
+       (when (and (:date1 compare) (:date2 compare))
+         [:div
+          [:h3 "Valittujen päivämäärien väliset muutokset"]
+          [comparison-date-changes compare]])]]]))
```

Comment:
This is ok, another alternative could be to have `comparison-date-changes`displayed all the time and just clear it's data model to have it displayed empty for a moment, if such UX is desired.
No need to change, as said :)